### ISSUE TITLE
Runtime TypeError getBigUint64 is not a function using Safari 14

### ISSUE THREAD

**Issue by pedelman** [timostamm/protobuf-ts#24]
Hi @timostamm, I have recently updated to Safari 14 and noticed the following error in my application which appears to be coming from the runtime. https://github.com/timostamm/protobuf-ts/blob/72866167d93aa0471da25dc82475014465a08847/packages/runtime/src/pb-long.ts#L159

```
Unhandled Promise Rejection: TypeError: O.getBigUint64 is not a function. (In 'O.getBigUint64(0,!0)', 'O.getBigUint64' is undefined)
```

Surprisingly, I didnt see this error in Safari 13, despite this method apparently not being supported. Im guessing the error was silently ignored?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView#Browser_compatibility

I did find some discussion on this topic already and it looks like it wouldnt be too bad to write a polyfill using JSBI. 

- https://programmer.help/blogs/how-to-write-uint64-data-to-array-buffer.html
- https://github.com/elk-chat/elk_web/blob/master/packages/sdk/bigint-buffer.ts
- https://github.com/GoogleChromeLabs/jsbi

#### System information
- macOS: `10.15.7`
- Safari: `Version 14.0 (15610.1.28.1.9, 15610)`

Let me know if there is any other information I can provide! Thanks again for putting together this project!

**Comment by timostamm** [comment]
Hi Patrick, 

unfortunately, BigInt can't be used in browsers because of Safari :-/

protobuf-ts supports two alternatives to the BigInt representation: JavaScript `number` and `string`.  Instead of `BigInt`, your fields will have the `number` or `string` type. The string representation is the most reliable and versatile. The easiest way to enable it is to set the plugin parameter `long_type_string`. See the [manual](https://github.com/timostamm/protobuf-ts/blob/v1.0.7/MANUAL.md#bigint-support) for details.

For arithmetics across browsers, you need a 3rd party lib. JSBI, or [long.js](https://github.com/dcodeIO/Long.js/). 

```typescript
// using long.js:
let a = Long.fromString(myMessage.int64Field)
let b = a.add(123);
myMessage.int64Field = b.toString();

// JSBI, very similar:
let a = JSBI.BigInt(myMessage.int64Field)
let b = a.add(123);
myMessage.int64Field = b.toString();
```

It thought about supporting JSBI as a representation and decided against, mostly because it would add a dependency to the runtime (at the moment, it has no deps at all). 


Anyways, the error message in Safari should be improved! What about: 

> BigInt unavailable. See https://bit.ly/3lpAaUM

Would this have helped you to find / fix the problem?

Not sure about Safari 13... my Version 13.1 (14609.1.20.111.8) shows the same type error when trying to call `getBigUint64`. To reproduce, I'd need some more info.

**Comment by pzeinlinger** [comment]
Hey @timostamm,
first thanks for your amazing work!!! :)

Using your suggested workaround long_type_string does not seem to work. The proto compiler does indeed replace all BigInts with string types but the error persists. Apparently it is thown by runtime-rpc:68/grpc-web-transport:142.

Any ideas?

**Comment by pedelman** [comment]
Hmm I have the `long_type_string` enabled and seeing this still.

Here are the options I a using from my build script.

```
npx protoc --ts_out some_dir/src --ts_opt long_type_string --ts_opt disable_service_client --ts_opt optimize_code_size --ts_opt generate_dependencies ...
```

Appreciate the prompt reply!

**Comment by pzeinlinger** [comment]
Using `long_type_string` I still have references to getBigUint64 left in the compiled code.

**Comment by timostamm** [comment]
Ok, this sounds like there is some code path that calls `PbUlong.toBigInt()`, even though code is generated for string representation. 

A stack trace would be super helpful. 

@pzeinlinger, are you on Safari 14 as well? 

I'll do an update to Safari 14 and try to get this reproduced...

**Comment by pzeinlinger** [comment]
Yes I am. Tried it on an iPad as well as Catalina.

**Comment by pedelman** [comment]
<img width="675" alt="Screen Shot 2020-11-12 at 2 45 37 PM" src="https://user-images.githubusercontent.com/955777/99005484-c2e05000-24f5-11eb-8abb-2b272d5e7459.png">
https://github.com/timostamm/protobuf-ts/blob/72866167d93aa0471da25dc82475014465a08847/packages/runtime/src/pb-long.ts#L149

**Comment by timostamm** [comment]
Thank you both for the details!  This is fixed in release v1.0.8.

Safari 14 adds the BigInt constructor function, but it does not implement the [DataView BigInt methods](https://tc39.es/ecma262/#sec-dataview.prototype.setbigint64). 

The fixed code if all necessary BigInt features are present. 

When running bigint code in Safari, the following error is raised: ![image](https://i.imgur.com/97ZIPWG.png)

**Comment by pedelman** [comment]
Thanks @timostamm!! Appreciate it!
